libzfs_core/_libzfs_core.py

```diff
@@ -0,0 +1,659 @@
+# Copyright 2015 ClusterHQ. See LICENSE file for details.
+
+"""
+Python wrappers for libzfs_core interfaces.
+
+As a rule, there is a Python function for each C function.
+The signatures of the Python functions generally follow those of the
+functions, but the argument types are natural to Python.
+nvlists are wrapped as dictionaries or lists depending on their usage.
+Some parameters have default values depending on typical use for
+increased convenience.  Output parameters are not used and return values
+are directly returned.  Error conditions are signalled by exceptions
+rather than by integer error codes.
+"""
+
+import errno
+import threading
+from . import exceptions
+from . import _error_translation as xlate
+from .bindings import libzfs_core
+from ._constants import MAXNAMELEN
+from ._nvlist import nvlist_in, nvlist_out
+
+
+def lzc_create(name, is_zvol = False, props = {}):
```

Comment:
Parameter defaults are only evaluated once.  The resulting object is re-used by every call to the function:

```py
>>> def f(g=[]):
...     g.append(3)
...     return g
... 
>>> f()
[3]
>>> f()
[3, 3]
>>> f()
[3, 3, 3]
>>> 
```

Thus it's typically a good idea to avoid using mutable objects as default values.  The typical idiom for this is:

```py
def f(g=None):
    if g is None:
        g = []
    g.append(3)
    return g
```